Switch from Travis-Ci to GitHub Actions

boesing opened the pull request:
<!--
Fill in the relevant information below to help triage your issue.

Pick the target branch based on the following criteria:
  * Documentation improvement: master branch
  * Bugfix: master branch
  * QA improvement (additional tests, CS fixes, etc.) that does not change code
    behavior: master branch
  * New feature, or refactor of existing code: develop branch

You MUST provide a signoff in your commits for us to be able to accept your
patch; you can do this by providing either the --signoff or -s flag when using
"git commit". Please see the project contributing guide and
https://developercertificate.org for details.
-->

|    Q          |   A
|-------------- | ------
| QA            | yes

### Description

As of the decision in one of the laminas TSC meetings, we are switching over to GHA.

boesing committed:
qa: replace `.travis.yml` with GHA workflow

Signed-off-by: Maximilian Bösing <2189546+boesing@users.noreply.github.com>

boesing committed:
qa: add `benchmark` script so CI matrix can work with it

Signed-off-by: Maximilian Bösing <2189546+boesing@users.noreply.github.com>

boesing committed:
qa: re-add `composer.lock` for CI pipeline

Using locked dependencies for codestyle and static analysis is better to avoid failing checks in unrelated PRs

Signed-off-by: Maximilian Bösing <2189546+boesing@users.noreply.github.com>

Ocramius commented:
On lowest dependencies:

```

1) LaminasTest\ServiceManager\LazyServiceIntegrationTest::testCanUseLazyServiceFactoryFactoryToCreateLazyServiceFactoryToActAsDelegatorToCreateLazyService
Error: Class "Zend\Code\Generator\ClassGenerator" not found

/github/workspace/vendor/ocramius/proxy-manager/src/ProxyManager/Generator/ClassGenerator.php:15
/github/workspace/vendor/ocramius/proxy-manager/src/ProxyManager/Factory/AbstractBaseFactory.php:104
/github/workspace/vendor/ocramius/proxy-manager/src/ProxyManager/Factory/AbstractBaseFactory.php:72
/github/workspace/vendor/ocramius/proxy-manager/src/ProxyManager/Factory/LazyLoadingValueHolderFactory.php:29
/github/workspace/src/Proxy/LazyServiceFactory.php:65
/github/workspace/src/ServiceManager.php:639
/github/workspace/src/ServiceManager.php:643
/github/workspace/src/ServiceManager.php:667
/github/workspace/src/ServiceManager.php:253
/github/workspace/test/LazyServiceIntegrationTest.php:137
```

boesing commented:
@Ocramius yup, thats due to the fact of this bug in laminas-dependency-plugin: 
laminas/laminas-dependency-plugin#28

boesing committed:
qa: require latest version of `laminas/laminas-dependency-plugin`

Fixes composer issues regarding PHP 8

Signed-off-by: Maximilian Bösing <2189546+boesing@users.noreply.github.com>

boesing commented:
Okay, I've fixed laminas/laminas-dependency-plugin#28 in `laminas/laminas-dependency-plugin` v2.1.2 and added it as minimum requirement to this package.
